src/main/scala/dsptools/counters/Counters.scala

```diff
@@ -164,11 +165,13 @@ abstract class Counter(countParams: CountParams) extends Module {
       //    else Mux(wrap,wrapTo,nextInSeq)
     }
     else {
-      if(wrap == Bool(true)) {
-        wrapTo
-      } else {
-        nextInSeq
+      val nextThing = Wire(Bool())
```

Comment:
Is this a compile-time optimization? There's a comparison between Chisel types (between `wrap` and `true.B` - though in the original there's a double-equals `==` Scala-land compare, which I'm unsure will work correctly) which would look like you intended to generate a hardware compare?